solarkraft/src/instrument.ts

```diff
@@ -287,16 +333,34 @@ export function tlaJsonOfNative(v: any, forceVec: boolean = false): any {
                 (key) => typeof key === 'string' && isTlaName(key)
             )
         ) {
+            const entries = Object.entries(v)
+            const flatEntries: any[] = entries.flat()
+            const childHints = entries
+                .map((arg) => {
+                    const k: string = arg[0]
+                    return [
+                        'Str',
+                        hint === undefined
+                            ? undefined
+                            : k in hint
+                              ? hint[k]
+                              : undefined,
```

Comment:
I think here you can 
```suggestion
                        (hint ?? {})[k]
```